apps/site/assets/ts/schedule/components/ScheduleFinder.tsx

```diff
@@ -0,0 +1,196 @@
+  <div className={`schedule-finder__select-container ${error && "error"}`}>
+    {children}
+    <div
+      className="schedule-finder__arrow"
+      dangerouslySetInnerHTML={{ __html: "&#xF107;" }}
+    />
+  </div>
+);
+
+interface ErrorMessageProps {
+  directionError: boolean;
+  originError: boolean;
+}
+
+const ErrorMessage = ({
+  directionError,
+  originError
+}: ErrorMessageProps): ReactElement<HTMLElement> | null => {
+  if (!directionError && !originError) {
+    return null;
+  }
+
+  const message =
+    directionError && originError
+      ? "direction and origin"
+      : directionError
+      ? "direction"
+      : "origin";
+
+  return (
+    <div className="error-container">
+      <span role="alert">Choose a value for {message}</span>
+    </div>
+  );
+};
+
+const ScheduleFinder = ({
+  directionDestinations,
+  directionNames,
+  stops
+}: Props): ReactElement<HTMLElement> => {
+  const submitButtonEl = React.useRef<HTMLInputElement>(null);
+
+  const [state, setState] = useState<State>({
+    selectedDirection: null,
+    selectedOrigin: null,
+    directionError: false,
+    originError: false
+  });
+
+  const handleClickSubmit = (e: SyntheticEvent): void => {
+    e.preventDefault();
+
+    if (state.selectedDirection === null || state.selectedOrigin === null) {
+      e.stopPropagation(); // prevent modal from opening
+      setState({
+        ...state,
+        directionError: state.selectedDirection === null,
+        originError: state.selectedOrigin === null
+      });
+      return;
+    }
+
+    setState({ ...state, directionError: false, originError: false });
+  };
+
+  const handleChangeDirection = (direction: SelectedDirection): void => {
+    setState({ ...state, selectedDirection: direction });
+  };
+
+  const handleChangeOrigin = (origin: SelectedOrigin): void => {
+    setState({ ...state, selectedOrigin: origin });
+  };
+
+  return (
+    <div className="schedule-finder">
+      <h2 className="h3 schedule-finder__heading">
+        {renderSvg("c-svg__icon", icon)} Schedule Finder
+      </h2>
+      <ErrorMessage
+        directionError={state.directionError}
+        originError={state.originError}
+      />
+      <div>
+        Choose a stop to get schedule information and real-time departure
+        predictions.
+      </div>
+      <SelectContainer error={state.directionError}>
+        <select
+          id="sf_direction_select"
+          className="schedule-finder__select"
+          onChange={e =>
+            handleChangeDirection(
+              e.target.value ? parseInt(e.target.value) : null
+            )
+          }
+          onKeyUp={e =>
+            handleReactEnterKeyPress(e, () => {
+              submitButtonEl!.current!.click();
+            })
+          }
+        >
+          <option value="">Choose a direction</option>
+          <option value="0">
+            {directionNames[0]}&nbsp;&nbsp;&nbsp;{directionDestinations[0]}
```

Comment:
This is harder to read 😬 , why not a template string?

Reply:
I tried it, but in FF I still saw crazy characters :/